src/main/java/org/folio/dew/batch/bursarfeesfines/TransferFeesFinesTasklet.java

```diff
@@ -2,34 +2,37 @@
 
 import java.util.List;
 import lombok.RequiredArgsConstructor;
-import org.apache.commons.collections4.CollectionUtils;
 import org.folio.dew.batch.ExecutionContextUtils;
 import org.folio.dew.batch.bursarfeesfines.service.BursarExportService;
-import org.folio.dew.domain.dto.Account;
-import org.folio.dew.domain.dto.bursarfeesfines.BursarJobPrameterDto;
+import org.folio.dew.domain.dto.BursarExportJob;
+import org.folio.dew.domain.dto.bursarfeesfines.AccountWithAncillaryData;
 import org.springframework.batch.core.StepContribution;
 import org.springframework.batch.core.configuration.annotation.StepScope;
 import org.springframework.batch.core.scope.context.ChunkContext;
 import org.springframework.batch.core.step.tasklet.Tasklet;
 import org.springframework.batch.repeat.RepeatStatus;
 import org.springframework.stereotype.Component;
 
-@RequiredArgsConstructor
 @Component
 @StepScope
+@RequiredArgsConstructor
 public class TransferFeesFinesTasklet implements Tasklet {
 
   private final BursarExportService exportService;
 
   @Override
   public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) {
-    List<Account> accounts = (List<Account>) ExecutionContextUtils.getExecutionVariable(contribution.getStepExecution(),
-        "accounts");
-    if (CollectionUtils.isNotEmpty(accounts)) {
-      exportService.transferAccounts(accounts,
-        (BursarJobPrameterDto) ExecutionContextUtils.getExecutionVariable(contribution.getStepExecution(), "bursarFeeFines"));
+    // from AccountItemReader
+    List<AccountWithAncillaryData> filteredAccounts = (List<AccountWithAncillaryData>) contribution.getStepExecution()
+      .getJobExecution()
+      .getExecutionContext()
+      .get("filteredAccounts");
+
+    if (filteredAccounts != null && !filteredAccounts.isEmpty()) {
```

Comment:
Method from Apache CollectionUtils.isNotEmpty can be useful here to replace 2 validations in if-statement.